### PR TITLE
DOC: update the pandas.DataFrame.notna and pandas.Series.notna docstring

### PR DESCRIPTION
**Pull request by datadonK23**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Two Validations for pandas.DataFrame.notna and pandas.Series.notna (shared docs).
```
################################################################################
###################### Docstring (pandas.DataFrame.notna) ######################
################################################################################

Detect existing (non-missing) values.

Return a boolean same-sized object indicating if the values are not NA.
Non-missing values get mapped to True. Characters such as empty
strings `''` or :attr:`numpy.inf` are not considered NA values
(unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
NA values, such as None or :attr:`numpy.NaN`, get mapped to False
values.

Returns
-------
bool of type DataFrame
    Mask of True/False values for each element in DataFrame that
    indicates whether an element is not an NA value

See Also
--------
DataFrame.notnull : alias of notna
DataFrame.isna : boolean inverse of notna
DataFrame.dropna : omit axes labels with missing values
notna : top-level notna

Examples
--------
Show which entries in a DataFrame are not NA.

>>> df = pd.DataFrame({'age': [5, 6, np.NaN],
...                    'born': [pd.NaT, pd.Timestamp('1939-05-27'),
...                             pd.Timestamp('1940-04-25')],
...                    'name': ['Alfred', 'Batman', ''],
...                    'toy': [None, 'Batmobile', 'Joker']})
>>> df
   age       born    name        toy
0  5.0        NaT  Alfred       None
1  6.0 1939-05-27  Batman  Batmobile
2  NaN 1940-04-25              Joker

>>> df.notna()
     age   born  name    toy
0   True  False  True  False
1   True   True  True   True
2  False   True  True   True

Show which entries in a Series are not NA.

>>> ser = pd.Series([5, 6, np.NaN])
>>> ser
0    5.0
1    6.0
2    NaN
dtype: float64

>>> ser.notna()
0     True
1     True
2    False
dtype: bool

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.notna" correct. :)
```

```
################################################################################
####################### Docstring (pandas.Series.notna)  #######################
################################################################################

Detect existing (non-missing) values.

Return a boolean same-sized object indicating if the values are not NA.
Non-missing values get mapped to True. Characters such as empty
strings `''` or :attr:`numpy.inf` are not considered NA values
(unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
NA values, such as None or :attr:`numpy.NaN`, get mapped to False
values.

Returns
-------
bool of type Series
    Mask of True/False values for each element in Series that
    indicates whether an element is not an NA value

See Also
--------
Series.notnull : alias of notna
Series.isna : boolean inverse of notna
Series.dropna : omit axes labels with missing values
notna : top-level notna

Examples
--------
Show which entries in a DataFrame are not NA.

>>> df = pd.DataFrame({'age': [5, 6, np.NaN],
...                    'born': [pd.NaT, pd.Timestamp('1939-05-27'),
...                             pd.Timestamp('1940-04-25')],
...                    'name': ['Alfred', 'Batman', ''],
...                    'toy': [None, 'Batmobile', 'Joker']})
>>> df
   age       born    name        toy
0  5.0        NaT  Alfred       None
1  6.0 1939-05-27  Batman  Batmobile
2  NaN 1940-04-25              Joker

>>> df.notna()
     age   born  name    toy
0   True  False  True  False
1   True   True  True   True
2  False   True  True   True

Show which entries in a Series are not NA.

>>> ser = pd.Series([5, 6, np.NaN])
>>> ser
0    5.0
1    6.0
2    NaN
dtype: float64

>>> ser.notna()
0     True
1     True
2    False
dtype: bool

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.notna" correct. :)
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
...
 
     _shared_docs['isna'] = """
+        Detect missing values.
+
         Return a boolean same-sized object indicating if the values are NA.
+        NA values, such as None or :attr:`numpy.NaN`, get mapped to True
+        values.
+        Everything else get mapped to False values. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Does this `attr` work? I don't know if its in our API docs. I think it'd be ok to jsut have unless you set \`\`pandas.options.mode.use_inf_as_na = True\`\`

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5476,14 +5526,63 @@ def isnull(self):
...
 
     _shared_docs['notna'] = """
-        Return a boolean same-sized object indicating if the values are
-        not NA.
+        Detect existing (non-missing) values.
+
+        Return a boolean same-sized object indicating if the values are not NA.
+        Non-missing values get mapped to True. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Same here about the pandas option.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
...
         Return a boolean same-sized object indicating if the values are NA.
+        NA values, such as None or :attr:`numpy.NaN`, get mapped to True
+        values.
+        Everything else get mapped to False values. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
+
+        Returns
+        -------
+        bool of type %(klass)s
```
Comment (villasv) [inline pandas/core/generic.py]: Is this type annotation right? It looks like `bool` is the return variable name (which is not needed).
https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-3-parameters

**Review by villasv — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (donK23) [reply pandas/core/generic.py]: We have seen :attr: references throughout other docstrings (frame.py). I can change it, thoughts?

File: pandas/core/generic.py (hunk 3 above)
Comment (donK23) [reply pandas/core/generic.py]: Not the name

Comment (villasv) [reply pandas/core/generic.py]: A complex type then? So it should be somehting like "dict of int", but it looks like it's inverted.

Comment (donK23) [reply pandas/core/generic.py]: We get a DataFrame or a Series  that contains boolean values. 
It feels like writing `DataFrame of bool`or `Series of bool` may be harder to userstand for new users. Any thoughts?

Comment (villasv) [reply pandas/core/generic.py]: The guidelines specify that complex types should be written like `list of bool`. Maybe you can just say that the return type is `DataFrame` and in the explanation specify that its dtype is bool.

File: pandas/core/generic.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/generic.py]: It's fine.

File: pandas/core/generic.py (hunk 3 above)
Comment (TomAugspurger) [reply pandas/core/generic.py]: I think the type should be just `%(klass)s`

```
%(klass)s
    Each element of the %(klass)s will be a boolean.
```

Comment (villasv) [reply pandas/core/generic.py]: Agreed, it's the closes to the guidelines.

File: pandas/core/generic.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/generic.py]: It's not the `attr` itself, but the fact there is nothing to link to I think?

I would rather make this ``` ``pandas.options.mode.use_inf_as_na = True`` ```

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
     # Action Methods
 
     _shared_docs['isna'] = """
+        Detect missing values.
+
         Return a boolean same-sized object indicating if the values are NA.
+        NA values, such as None or :attr:`numpy.NaN`, get mapped to True
+        values.
+        Everything else get mapped to False values. Characters such as empty
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: get -> gets ?

**Review by jorisvandenbossche — COMMENTED** [review]
Nice docstrings! Added a very minor comment

File: pandas/core/generic.py (hunk 4 above)
Comment (donK23) [reply pandas/core/generic.py]: Sry, not my language. Fixed the typos and changed the link as proposed in .notna docstring.

**Review by TomAugspurger — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@donK23 Thanks a lot!

If possible, welcome to check the online docs in a few hours to see if everything is looking OK: http://pandas-docs.github.io/pandas-docs-travis/generated/pandas.DataFrame.notna.html#pandas.DataFrame.notna

**Pull request merged**